index.js

```diff
@@ -1,41 +1,31 @@
 'use strict';
-var fs = require('fs');
-
-module.exports = function (filepath, pos, len, cb) {
-	var buf = new Buffer(len);
-
-	fs.open(filepath, 'r', function (err, fd) {
-		if (err) {
-			cb(err);
-			return;
-		}
-
-		fs.read(fd, buf, 0, len, pos, function (err, bytesRead, buf) {
-			if (err) {
-				cb(err);
-				return;
-			}
-
-			fs.close(fd, function (err) {
-				if (err) {
-					cb(err);
-					return;
-				}
+const fs = require('fs');
+const pify = require('pify');
+
+module.exports = (filepath, pos, len) => {
+	const buf = new Buffer(len);
+
+	return pify(fs.open)(filepath, 'r').then(fd => {
+		return pify(fs.read, {multiArgs: true})(fd, buf, 0, len, pos).then(res => {
```

Comment:
If we store `fd` outside, we also need to store `res` outside because we're doing `fs,close` before actually doing anything with `res`.